### PR TITLE
Add dependencies for Arch Linux

### PR DESCRIPTION
**Pull request by JamesTobin**
This pull request adds Arch Linux dependencies (including base-devel). Arch Linux is quite popular, especially among Minetest players, as are derivatives such as Manjaro.

## To do

This PR is Ready for Review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,6 +147,10 @@ For Debian/Ubuntu users:
 For Fedora users:
 
     sudo dnf install make automake gcc gcc-c++ kernel-devel cmake libcurl-devel openal-soft-devel libvorbis-devel libXxf86vm-devel libogg-devel freetype-devel mesa-libGL-devel zlib-devel jsoncpp-devel irrlicht-devel bzip2-libs gmp-devel sqlite-devel luajit-devel leveldb-devel ncurses-devel doxygen spatialindex-devel bzip2-devel
+    
+For Arch users:
+
+    sudo pacman -S base-devel gnucurl libcurl-gnutls cmake openal libvorbis vorbis-tools libxxf86vm irrlicht libpng libpng png++ glu mesa sqlite libspatialite libogg libvorbis openal freetype2 jsoncpp gmp luajit leveldb ncurses doxygen
```
Comment (SmallJoker) [inline README.md]: Contains a few duplicates, such as openal, libpng and libvorbis

**Comment by sfan5** [comment]
> gnucurl

this does not exist

> vorbis-tools

not needed, libvorbis + libogg is enough

> png++ glu mesa libspatialite

we don't use any of these

> doxygen

most users won't need this so I'd leave it off the list

**Comment by JamesTobin** [comment]
I meant gnurl instead of gnucurl, my bad

**Comment by JamesTobin** [comment]
updated deps

**Comment by sfan5** [comment]
Minetest does not use `gnurl` either.

**Comment by JamesTobin** [comment]
I believe it replaces the standard libcurl

**Comment by sfan5** [comment]
It does not, but even if it did: Why is the standard libcurl not okay?

**Comment by JamesTobin** [comment]
there's no libcurl package. maybe libcurl-gnutls is enough?

**Comment by sfan5** [comment]
`libcurl-gnutls` provides libcurl and is what Minetest will use.

**Comment by JamesTobin** [comment]
ok

**Comment by JamesTobin** [comment]
done

**Pull request merged**